exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/common/HashPartition.java

```diff
@@ -0,0 +1,579 @@
+    return inMemoryBatchStats.size();
+  }
+
+  @Override
+  public boolean isSpilled() {
+    return isSpilled;
+  }
+
+  @Override
+  public long getNumInMemoryRecords() {
+    return numInMemoryRecords;
+  }
+
+  @Override
+  public long getInMemorySize() {
+    return partitionInMemorySize;
+  }
+
+  public String getSpillFile() {
+    return spillFile;
+  }
+
+  public int getPartitionBatchesCount() {
+    return partitionBatchesCount;
+  }
+  public int getPartitionNum() {
+    return partitionNum;
+  }
+
+  /**
+   * Close the writer without deleting the spill file
+   */
+  public void closeWriter() { // no deletion !!
+    closeWriterInternal(false);
+    processingOuter = true; // After the spill file was closed
+  }
+  /**
+   * If exists - close the writer for this partition
+   *
+   * @param doDeleteFile Also delete the associated file
+   */
+  private void closeWriterInternal(boolean doDeleteFile) {
+    try {
+      if ( writer != null ) {
+        spillSet.close(writer);
+      }
+      if ( doDeleteFile && spillFile != null ) {
+        spillSet.delete(spillFile);
+      }
+    } catch (IOException ioe) {
+      throw UserException.resourceError(ioe)
+        .message("IO Error while closing %s spill file %s",
+          doDeleteFile ? "and deleting" : "",
+          spillFile)
+        .build(logger);
+    }
+    spillFile = null;
+    writer = null;
+    partitionBatchesCount = 0;
+  }
+
+  /**
+   * Creates the hash table and join helper for this partition.
+   * This method should only be called after all the build side records
+   * have been consumed.
+   */
+  public void buildContainersHashTableAndHelper() throws SchemaChangeException {
+    if ( isSpilled ) { return; } // no building for spilled partitions
+    containers = new ArrayList<>();
+    hashTable.updateInitialCapacity((int) getNumInMemoryRecords());
+    for (int curr = 0; curr < partitionBatchesCount; curr++) {
+      VectorContainer nextBatch = tmpBatchesList.get(curr);
+      final int currentRecordCount = nextBatch.getRecordCount();
+
+      // For every incoming build batch, we create a matching helper batch
+      hjHelper.addNewBatch(currentRecordCount);
+
+      // Holder contains the global index where the key is hashed into using the hash table
+      final IndexPointer htIndex = new IndexPointer();
+
+      assert nextBatch != null;
+      assert probeBatch != null;
+
+      hashTable.updateIncoming(nextBatch, probeBatch );
+
+      // IntVector HV_vector = (IntVector) nextBatch.getValueVector(rightHVColPosition).getValueVector();
+      IntVector HV_vector = (IntVector) nextBatch.getLast();
+
+      for (int recInd = 0; recInd < currentRecordCount; recInd++) {
+        int hashCode = HV_vector.getAccessor().get(recInd);
+        try {
+          hashTable.put(recInd, htIndex, hashCode);
+        } catch (RetryAfterSpillException RE) {
+          throw new OutOfMemoryException("HT put");
+        } // Hash Join can not retry yet
```

Comment:
Please remove comment "// Hash Join can not retry yet". HashJoin uses a proactive memory approach so we don't need to do retrys.

Reply:
Changed to "Hash Join does not retry"